enctests/testframework/main.py

```diff
@@ -303,6 +319,126 @@ def vmaf_compare(source_clip, test_ref, testname):
+    extract_template = comparisontest_info.get("extracttemplate", default_extract_template)
+
+    # Allow a different image to be compared with, useful for 422 or 420 encoding.
+    sourcepng = comparisontest_info.get("compare_image", source_path.as_posix())
+
+    distortedbase, distortedext = os.path.splitext(distorted)
+    distortedpng = os.path.join(os.path.dirname(distorted), distortedbase + ".png")
+
+    extractcmd = extract_template.format(newfile=distorted, newpngfile=distortedpng)
+    print("About to extract with cmd:", extractcmd)
+    result = {'success': False,
+              'result': "undefined"
+    }
+    cmdresult = subprocess.call(shlex.split(extractcmd))
+    if cmdresult != 0:
+        result['result'] = "Unable to extract file for test"
+    else:
+        cmd = apptemplate.format(originalfile=source_path, newfile=distortedpng)
+        print("Idiff command:", cmd)
+        
+        output = subprocess.run(shlex.split(cmd), check=False, stdout=subprocess.PIPE).stdout
+        lines = output.decode("utf-8").splitlines()
+        if len(lines) < 2:
+            result['result'] = "Unable to run idiff"
+            result['success'] = False
+        else:
+            result = {'success': lines[-1] == "PASS", 'result': lines[-1]}
+            for line in lines[:-1]:
+                if " = " not in line:
+                    continue
+                key, value = line.split(" = ")
+                key = key.strip().replace(" ", "_").lower()
+                if key == "max_error":
+                    value, location = value.split(" @ ")
+                    result['max_location'] = location
+                else:
+                    value = value.strip()
+                result[key] = float(value)
+    enc_meta = get_test_metadata_dict(test_ref)
+    enc_meta['results'].update(result)
+
+def assertresults_compare(source_clip, test_ref, testname, comparisontest_info):
+    """
+    Check the results of the tests against known values (or value ranges).
+    We assume that we have already run some tests, and just want to check that the values are good.
+    
+    source_clip -- The original clip that the new media is based on. (not used)
+    test_ref    -- The generated movie we are testing.(not used)
+    testname    -- The testname we are running.
+    comparisontestinfo -- other parameters that can be used to configure the test.
+
+    testresult - Was the test able to run (Completed = Yes)
+    success - Boolean, was the test a success. 
+    """
+    tests = comparisontest_info.get("tests", [])
+    enc_meta = get_test_metadata_dict(test_ref)
+    result = enc_meta['results']
+    resultstatus = True
+    for test in tests:
+        if "assert" not in test:
+            print("WARNING: no test to run in test:", test, " expecting a field called assert with the test type.")
```

Comment:
```suggestion
            print(f"WARNING: no test to run in test: {test} expecting a field called assert with the test type.")
```
We should try to use f-strings when possible